managers/tests/test_views.py

```diff
@@ -225,3 +229,89 @@ def test_delete_project_function_view_should_not_delete_project_on_non_admin_req
+
+    def test_manage_task_activities_in_project_view_should_raise_404_for_manager_not_of_current_project(self):
+        other_manager = ManagerUserFactory()
+        self.client.force_login(other_manager)
+
+        response = self.client.get(self.url2)
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_manage_task_activities_in_project_view_should_get_200_when_user_is_manager_of_current_project(self):
+        project_manager = ManagerUserFactory()
+        self.project.managers.add(project_manager)
+        self.client.force_login(project_manager)
+
+        response = self.client.get(self.url2)
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_manage_task_activities_in_project_view_should_redirect_on_employee_get(self):
+        user_employee = UserFactory()
+        self.project.members.add(user_employee)
+        self.client.force_login(user_employee)
+
+        response = self.client.get(self.url2)
+
+        self.assertEqual(response.status_code, 302)
+
+    def test_manage_task_activities_view_should_show_all_task_activities_in_project_and_available_to_add(self):
+        self.assertCountEqual(
+            self.project.project_activities.all(), TaskActivityType.objects.filter_default_task_activities()
+        )
+        self.assertNotIn(self.not_default_task_activity, self.project.project_activities.all())
+
+        response = self.client.get(self.url2)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.default1_task_activity.name)
+        self.assertContains(response, self.not_default_task_activity.name)
+
+    def test_manage_task_activities_view_should_add_new_task_activity_on_post_and_add_it_to_project(self):
+        task_activity_name = "New activity"
+        response = self.client.post(self.url2, data={"name": task_activity_name})
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, task_activity_name)
+        task_activity = TaskActivityType.objects.get(name=task_activity_name)
+        self.assertIn(task_activity, self.project.project_activities.all())
+
+    def test_manage_task_activities_view_should_add_existing_task_activity_to_project_on_add_to_project_function(self):
+        length_of_default_task_activities = TaskActivityType.objects.filter_default_task_activities().count()
+        self.assertEqual(self.project.project_activities.all().count(), length_of_default_task_activities)
+
+        response = self.client.get(self.url2 + f"?task_activities={self.not_default_task_activity.pk}")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.not_default_task_activity.name)
+        self.assertIn(self.not_default_task_activity, self.project.project_activities.all())
+        self.assertNotEqual(self.project.project_activities.all().count(), length_of_default_task_activities)
```

Comment:
I always prefer and recommend to checking if something is equal to something instead of not equal if you have opportunity to this, because 1 won't be equal to 2, but also won't be equal to -42109387123812.

Reply:
Yes, I that it is not precission enough, but I wanted to check if task activity list is different when I add new task than before when it contain only default tasks, however I will change it ASAP